sympy/assumptions/ask.py

```diff
@@ -729,6 +729,44 @@ def is_true(self):
         """
         return Predicate('is_true')
 
+    # XXX: empty_matrix or just empty?
+    @predicate_memo
+    def empty_matrix(self):
+        """
+        A matrix having at least one dimension equal to zero.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, MatrixSymbol
+        >>> X = MatrixSymbol('X', 0, 0)
+        >>> Y = MatrixSymbol('Y', 2, 0)
+        >>> Z = MatrixSymbol('Z', 2, 2)
+        >>> ask(Q.empty_matrix(X))
+        True
+        >>> ask(Q.empty_matrix(Z*Y))
+        True
+        >>> ask(Q.empty_matrix(Z))
+        False
+        """
+        return Predicate('empty_matrix')
+
+    # XXX: is scalar_matrix predicat needed?
+    @predicate_memo
+    def scalar_matrix(self):
+        """
+        A 1-by-1 matrix.
+        """
+        return Predicate('scalar_matrix')
+
+    # XXX: is zero_matrix predicat needed?
```

Comment:
We probably eventually want a more general way to make assumptions about all the elements of a matrix (like `Q.elements(Q.zero)` or something like that), but just having this is fine for now (even if we don't add it, 0 matrix is quite common, and I don't like using Q.zero for it). 

Then again, it might be worth moving this to a separate PR. There are some matrix facts that use Q.zero and we should probably make them use Q.zero_matrix.